pkg/executor/adapter.go

```diff
@@ -555,7 +555,12 @@ func (a *ExecStmt) Exec(ctx context.Context) (_ sqlexec.RecordSet, err error) {
 			return nil, err
 		}
 		if len(switchGroupName) > 0 {
-			stmtCtx.ResourceGroupName = switchGroupName
+			group, err := rm.ResourceGroupCtl.GetResourceGroup(switchGroupName)
```

Comment:
BTW, Have we already checked when setting up a non-existent group?

Reply:
Considering that the records of `QUERY WATCH` and the resource group are decoupled, there is currently no check on the existence of the resource group when setting up. This allows for a more flexible configuration of `QUERY WATCH` and simplifies the code. Here are a few typical scenarios: we can set the corresponding resource group after adding `SWITCH_GROUP`; or after deleting a resource group, the corresponding runaway statements will not fail immediately.